torch/quantization/fuse_modules.py

```diff
@@ -52,44 +61,68 @@ def fuse_conv_bn_relu(conv, bn, relu):
         return torch_fused.ConvReLU2d(
             torch.nn.utils.fusion.fuse_conv_bn_eval(conv, bn), relu)
 
+def _get_module(model, submodule_key):
+    submodule = ''
+    tokens = submodule_key.split('.')
+    cur_mod = model
+    for s in tokens:
+        if s.isdigit():
+            cur_mod = cur_mod[int(s)]
+        else:
+            cur_mod = getattr(cur_mod, s)
+    return cur_mod
+
+# Generalization of setattr
+def _set_module(model, submodule_key, module):
+    submodule = ''
+    tokens = submodule_key.split('.')
+    sub_tokens = tokens[:-1]
+    cur_mod = model
+    prev_mod = model
+    for s in sub_tokens:
+        if s.isdigit():
+            cur_mod = cur_mod[int(s)]
+        else:
+            cur_mod = getattr( cur_mod, s)
+
+    if tokens[-1].isdigit():
+        cur_mod[int(tokens[-1])] = module
+    else:
+        setattr(cur_mod, tokens[-1], module)
 
-def _fuse_modules(model, named_module_dict, modules_to_fuse, fuser_func=None):
-    assert(len(modules_to_fuse) == 2 or len(modules_to_fuse) == 3),\
-        "Can fuse only 2 or 3 modules."
+def _fuse_modules(model, modules_to_fuse, fuser_func=None):
 
     OP_LIST_TO_FUSER_FUNC = {
         (torch.nn.Conv2d, torch.nn.BatchNorm2d): fuse_conv_bn,
-        (torch.nn.Conv2d, torch.nn.BatchNorm2d, torch.nn.ReLU): fuse_conv_bn_relu
+        (torch.nn.Conv2d, torch.nn.BatchNorm2d, torch.nn.ReLU): fuse_conv_bn_relu,
+        (torch.nn.Conv2d, torch.nn.ReLU): fuse_conv_relu,
+        (torch.nn.Linear, torch.nn.ReLU): fuse_linear_relu
     }
 
     mod = []
-    parent_mod = []
-    for i in range(len(modules_to_fuse)):
-        parent_module_name = '.'.join(modules_to_fuse[i].split('.')[:-1])
-        mod.append(named_module_dict[modules_to_fuse[i]])
-        parent_mod.append(named_module_dict.get(parent_module_name, model))
+    for item in modules_to_fuse:
+        mod.append(_get_module(model, item))
 
-    new_mod = mod[0]
     if fuser_func is None:
         types = tuple(type(m) for m in mod)
         fuser_func = OP_LIST_TO_FUSER_FUNC.get(types, None)
```

Comment:
I'd pull this logic into something like 'fuse_known_modules' and pass it as a default to `fuser_func` - it's clear this way and makes it easier to extend functionality